pkg/deployment/deployment.go

```diff
@@ -315,6 +309,54 @@ func (builder *Builder) WithServiceAccountName(serviceAccountName string) *Build
 	return builder
 }
 
+// WithVolume sets the Volume on the deployment definition.
+func (builder *Builder) WithVolume(volume *coreV1.Volume) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Configuring volume %v on deployment %s in namespace %s",
+		volume, builder.Definition.Name, builder.Definition.Namespace)
+
+	if volume == nil {
+		glog.V(100).Infof("The 'volume' of the deployment is empty")
+
+		builder.errorMsg = "'volume' parameter is empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	mountConfig := coreV1.VolumeMount{Name: volume.Name, MountPath: volume.HostPath.Path, ReadOnly: false}
```

Comment:
```suggestion
	mountConfig := coreV1.VolumeMount{Name: volume.Name, MountPath: mountPath, ReadOnly: false}
```

Reply:
I don't understand why?

Reply:
you told me do not change type of volume to the volumeName. If it stay to be volume *coreV1.Volume object, so we have all inside

Reply:
volume.HostPath.Path is a required value

Reply:
Refer to my comment here https://github.com/openshift-kni/eco-goinfra/pull/165#discussion_r1385386274

Reply:
I mentioned the expected signature for the method here https://github.com/openshift-kni/eco-goinfra/pull/165#discussion_r1367327935

Reply:
> volume.HostPath.Path is a required value

Not sure where you're seeing that it is a required value for [VolumeSource](https://pkg.go.dev/k8s.io/api/core/v1#VolumeSource) resources. Can you share?